### ISSUE TITLE
New markdown link checker and linter are overly sensitive

### ISSUE THREAD

**Issue by samodell** [knative/docs#750]
Many PRs in the Docs repo right now have build test failures when it seems like they shouldn't.

Causes:
* Oversensitive link verification, such as #644 (more details in the comments of that PR; but to summarize, one link is valid and causing a failure, and one link is to a http://localhost URL, which we should probably exclude from link checks)

* Unknown: I can't see a root cause for some failures, such as #747 
Error logs are vague:
I0110 23:28:45.331] ============================
I0110 23:28:45.332] ==== BUILD TESTS FAILED ====
I0110 23:28:45.332] ============================
W0110 23:28:45.363] Run: ('/workspace/./test-infra/jenkins/../scenarios/../hack/coalesce.py',)
E0110 23:28:45.363] Command failed
I0110 23:28:45.363] process 643 exited with code 1 after 0.7m
E0110 23:28:45.363] FAIL: pull-knative-docs-build-tests

CC @srinivashegde86  @jessiezcc  @adrcunha

**Comment by adrcunha** [comment]
About the links, yes, the checker might be over sensitive. Probably it doesn't like an address without protocol, and I'm pretty sure localhost:4000 is not returning content at the time of checking. :)

About the second, scroll up the logs and you'll see:

```
I0110 23:28:24.004] ------------------------------------
I0110 23:28:24.005] ---- Linting the markdown files ----
I0110 23:28:24.005] ------------------------------------
I0110 23:28:26.564] serving/samples/helloworld-go/README.md:141: MD007 Unordered list indentation
I0110 23:28:26.565] 
I0110 23:28:26.565] A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
```

**Comment by samodell** [comment]
@adrcunha  with #747, I played around with the list formatting and wasn't able to satisfy the linter. The docs for the linter say that you must use 2 spaces for indented items in lists, but when I used that level of indentation, it broke the numbered list the unordered list is under by making the next numbered item in the list start over at 1. 

I think this linter might not be a great fit for our repo. Is there a way to disable the Markdown linter but keep the link checking capabilities?
